### PR TITLE
checking send limits for bf

### PR DESCRIPTION
**Pull request by husobee**
### Summary

Check the userdrain send limit for bitflyer.

### Type of change ( select one )

- [ ] Product feature
- [x] Bug fix
- [ ] Performance improvement
- [ ] Refactor
- [ ] Other

### Tested Environments

- [ ] Development
- [ ] Staging
- [ ] Production

### Before submitting this PR:

- [ ] Does your code build cleanly without any errors or warnings?
- [ ] Have you used auto closing keywords?
- [ ] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [ ] Have you squashed all intermediate commits?
- [ ] Is there a clear title that explains what the PR does?
- [ ] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security / privacy review if needed
- [ ] Have you performed a self review of this PR?

### CODE REVIEW

File: promotion/drain.go [hunk 1]
```diff
@@ -242,8 +242,26 @@ func (service *Service) RedeemAndTransferFunds(ctx context.Context, credentials
...
+
 		totalF64, _ := total.Float64()
 
+		// get quote, make sure we dont go over 100K JPY
+		quote, err := service.bfClient.FetchQuote(ctx, "BAT_JPY", false)
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch bitflyer quote")
+		}
+
+		JPYLimit := decimal.NewFromFloat(100000.0)
```
Comment (maikelmclauflin) [inline promotion/drain.go:254]: lets remove the `.0` golang will still use it

File: promotion/drain.go [hunk 2]
```diff
@@ -242,8 +242,26 @@ func (service *Service) RedeemAndTransferFunds(ctx context.Context, credentials
...
+		var overLimitErr error
+
+		totalJPYTransfer := total.Mul(quote.Rate)
+
+		if totalJPYTransfer.GreaterThan(JPYLimit) {
+			over := JPYLimit.Sub(totalJPYTransfer).String()
+			totalF64, _ = JPYLimit.Div(quote.Rate).Floor().Float64()
+			overLimitErr = fmt.Errorf("transfer is over 100K JPY by %s", over)
```
Comment (maikelmclauflin) [inline promotion/drain.go:262]: lets add the rate to this error string as well

**Review by maikelmclauflin — CHANGES_REQUESTED** [review]

File: promotion/drain.go (hunk 1 above)
Comment (husobee) [reply promotion/drain.go]: resolved in https://github.com/brave-intl/bat-go/pull/727/commits/db8854558cbde9dee599f3a5c0df5995b1ec42da

File: promotion/drain.go (hunk 2 above)
Comment (husobee) [reply promotion/drain.go]: resolved in https://github.com/brave-intl/bat-go/pull/727/commits/db8854558cbde9dee599f3a5c0df5995b1ec42da

**Review by maikelmclauflin — APPROVED** [review]

**Pull request merged**